atom-publisher-lib/src/main/scala/com/gu/atom/data/DynamoDataStore.scala

```diff
@@ -46,13 +43,15 @@ abstract class DynamoDataStore[D : ClassTag : DynamoFormat]
       case None => Left(IDNotFound)
     }
 
-  def createAtom(atom: Atom): DataStoreResult[Unit] = createAtom(DynamoCompositeKey(atom.id), atom)
+  def createAtom(atom: Atom): DataStoreResult[Atom] = createAtom(DynamoCompositeKey(atom.id), atom)
 
-  def createAtom(dynamoCompositeKey: DynamoCompositeKey, atom: Atom): DataStoreResult[Unit] =
+  def createAtom(dynamoCompositeKey: DynamoCompositeKey, atom: Atom): DataStoreResult[Atom] =
     if (get(uniqueKey(dynamoCompositeKey)).isDefined)
       fail(IDConflictError)
-    else
-      succeed(put(atom))
+    else {
+      put(atom)
```

Comment:
It feels strange that we call DynamoDB, but we do nothing with the result.
In case of failure, dynamo throws [exceptions](http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/CommonErrors.html) (in the docs you can find specific errors to every operation). I'd say it might be useful to catch them, even if it rarely fails.